XpringKit/DefaultXpringClient.swift

```diff
@@ -1,3 +1,5 @@
+import Foundation
```

Comment:
What is `Foundation`?

If this was JS code, I would say this looks like an unused import, but idk how this works in Swift-land.

Reply:
All Swift programs import the [Swift Standard Library](https://developer.apple.com/documentation/swift/swift_standard_library/) by default. This lets us use things like numbers or Strings. We haven't needed to import anything before now because we're not using anything more advanced. 

[Foundation](https://developer.apple.com/documentation/foundation) is a core framework Apple provides. In it, there is a `Data` type, which is an opaque wrapper for a collection of bytes. A data object is made on line 84  which takes the array of bytes (typed as '[Uint8]` in swift) returned from `toHex()` to a `Data`.
---
Since you piqued my curiousity, I went out and found out to my surprise that [class member imports are possible](https://nshipster.com/import/) even though I've *never* seen Swift code that uses them.  I googled 'Swift style guide' and of the top 3 hits, [LinkedIn's guide](https://github.com/linkedin/swift-style-guide) makes no mention of imports, whereas Ray Wenderlich's (a well known tutorial site in the swift community) [demonstrates whole module imports](https://github.com/raywenderlich/swift-style-guide#minimal-imports) but does not explicitly recommend them. [Google's guide](https://google.github.io/swift/#import-statements) is explicit about this, saying:

> Imports of whole modules are preferred to imports of individual declarations or submodules.
There are a number of reasons to avoid importing individual members:
> - There is no automated tooling to resolve/organize imports.
> - Existing automated tooling (such as Xcode’s migrator) are less likely to work well on code that imports individual members because they are considered corner cases.
> - The prevailing style in Swift (based on official examples and community code) is to import entire modules.

Reply:
Great explanation. I feel like I understand Swift a little better now!